### PR TITLE
fix(svm): derive composite ISM recipient from warp router

### PR DESCRIPTION
**Pull request by xeno097**
### Description

- Deferred artifact-API ISM deployment until the warp router exists, then resolved contextual ISM data from that router before attachment.
- Traversed nested routing artifacts and every composite ISM container so `rateLimited.recipient` is resolved and validated throughout the tree.
- Preserved NEW, DEPLOYED, and UNDERIVED artifact boundaries; reused DEPLOYED roots unchanged and rejected invalid NEW descendants.
- Rejected composite rate limiters anywhere in mailbox default ISM trees.
- Added schema, writer, validation, and Sealevel CLI E2E coverage.

### Drive-by changes

None.

### Related issues

None.

### Backward compatibility

Yes. Existing read-back configs retain explicit recipients. New warp deployments may omit `rateLimited.recipient`; it is derived from the deployed router. EVM and Tron keep their legacy deployment path.

### Testing

- Lint and typechecks: provider-sdk, deploy-sdk, sdk, svm-sdk, CLI
- 444 focused provider/deploy/sdk/SVM unit tests
- `CLI_E2E_TEST=warp-composite-ism pnpm -C typescript/cli test:sealevel:e2e` — 3 passing
- `CLI_E2E_TEST=warp-fee pnpm -C typescript/cli test:sealevel:e2e` — 4 passing
- `CLI_E2E_TEST=warp-apply-ism-add-test-ism pnpm -C typescript/cli test:starknet:e2e` — 1 passing

### CODE REVIEW

File: typescript/deploy-sdk/src/warp/warp-writer.ts [hunk 1]
```diff
@@ -169,17 +177,13 @@ export class WarpTokenWriter
...
-    // owner from the start rather than via a signer-as-initial-owner /
-    // TransferOwnership dance.
+    // Deploy warp WITHOUT an ISM or fee. Both are attached post-warp, and NEW
+    // artifacts are deployed first once the router address is available.
     const rawArtifact: ArtifactNew<RawWarpArtifactConfig> = {
       artifactState: ArtifactState.NEW,
       config: {
         ...config,
-        interchainSecurityModule: onChainIsmArtifact,
+        interchainSecurityModule: undefined,
```
Comment (paulbalaji) [inline typescript/deploy-sdk/src/warp/warp-writer.ts:190]: [P2] Keep signer ownership until the deferred ISM is attached

For a direct `createWarpTokenWriter(...).create(...)` call on Starknet with `config.owner !== signer.getSignerAddress()` and a custom ISM, the raw writer still receives the final owner. `StarknetWarpTokenWriterBase.create()` calls `applyPostCreateConfig()`, which transfers ownership before returning (`typescript/starknet-sdk/src/warp/token-artifact-manager.ts:406-415`). The new attachment step then submits the owner-gated ISM update using the original signer, so creation fails after deployment and leaves the requested ISM unattached. Previously the raw writer attached the ISM before transferring ownership.

The high-level CLI masks this by supplying the signer as an intermediate owner, but this factory is also a public deploy-sdk API. Retain signer ownership through attachment and transfer to the configured owner last; add backend coverage with distinct signer/owner addresses.

**Review by paulbalaji — COMMENTED** [review]
Reviewed at `e48bb99934734e89c7c2ef4cce0c2c546bf22206`. One P2 in the deferred ISM attachment path: direct Starknet writer deployments can transfer ownership before attaching the requested ISM. Details inline.

Reconciled existing discussion and review threads; no prior actionable feedback to duplicate. Independent review included Muse and five completed Kimi K3 swarm lanes, with findings checked against source. GLM-5.3 coverage was partial: stopped after repeated investigation of a pre-existing edge case, without a final verdict.

Validation: provider-sdk typecheck, lint and all 173 unit tests passed locally; `git diff --check` passed. Current CI checks pass (the earlier title-check failure has a successful rerun). Backend E2E tests were not rerun locally; broader SDK/deploy-sdk test attempts lacked package dependencies.

File: typescript/deploy-sdk/src/warp/warp-writer.ts (hunk 1 above)
Comment (xeno097) [reply typescript/deploy-sdk/src/warp/warp-writer.ts:190]: Fixed in `546832ba59`. The generic writer now raw-creates with the signer as temporary owner, then passes the deferred artifacts and configured final owner together through the regular update path. Backend ownership transactions remain last. Added distinct signer/owner coverage for both owner-only creation and deferred ISM attachment.

Comment (xeno097) [reply typescript/deploy-sdk/src/warp/warp-writer.ts:190]: CI is now fully green on `824f59eefb`. Thanks again for catching the ownership-ordering issue.

**Review by paulbalaji — APPROVED** [review]
Rereviewed at `824f59eefb0752d122f800cde98781d3d649473a`, including all changes since my previous review and the author replies. No new blocking findings.

The prior P2 is addressed: raw creation retains signer ownership, and the regular update attaches the requested ISM/fee before transferring to the configured owner. Owner-only creation is also covered. Checked actual backend transaction ordering, including Starknet, and the test-only RPC/explorer changes; an independent source rereview agreed.

Validation: all 51 warp-writer tests passed locally, including both new distinct signer/owner regressions; deploy-sdk lint and cumulative `git diff --check` passed. Current CI: 121 successful checks, 16 skipped, none failing or pending. The new regression tests mock backend transactions; backend E2E tests were not rerun locally.
